src/Workspaces/Remote/ServiceHub/Services/RemoteSymbolSearchUpdateEngine.cs

```diff
@@ -50,6 +51,28 @@ public Task UpdateContinuouslyAsync(string sourceName, string localSettingsDirec
             return serializedResults;
         }
 
+        private class ProgressService : ISymbolSearchProgressService
```

Comment:
this runs on the remote (non-VS) side (hence RemoteSymbolSearchUpdateEngine and Microsoft.CodeAnalysis.Remote).  The idea here is that when the update-engine in the remote side calls the async progress methods on ISymbolSearchProgressService , we will then remote those over to VS to do the progress work on the VS side. 

In this case "_service.Rpc.InvokeAsync" ends up calling over onto the VS side.  

(i agree this is all fairly confusing).